src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +244,56 @@ pub fn register(registry: &mut FunctionRegistry) {
         |map, _| map.len() as u64,
     );
 
+    registry.register_2_arg::<EmptyMapType, EmptyArrayType, EmptyMapType, _, _>(
+        "map_delete",
+        |_, _, _| FunctionDomain::Full,
+        |_, _, _| (),
+    );
+
+    registry.register_function_factory("map_delete", |_, arg_types| {
+        if arg_types.len() != 2 {
+            return None;
+        }
+
+        let (key_type, value_type) = match arg_types.first()? {
+            DataType::Map(box DataType::Tuple(type_tuple)) => {
+                if type_tuple.len() == 2 {
+                    (type_tuple[0].clone(), type_tuple[1].clone())
+                } else {
+                    return None;
+                }
+            }
+            _ => {
+                return None;
+            }
+        };
+
+        let key_array_type = match arg_types.get(1)? {
+            DataType::Array(box key_type) => key_type.clone(),
+            _ => {
+                return None;
+            }
+        };
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_delete".to_string(),
+                args_type: vec![
+                    DataType::Map(Box::new(DataType::Tuple(vec![
+                        key_type.clone(),
+                        value_type.clone(),
+                    ]))),
+                    DataType::Array(Box::new(key_array_type)),
+                ],
+                return_type: DataType::Map(Box::new(DataType::Tuple(vec![key_type, value_type]))),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(|_, _| FunctionDomain::Full),
```

Comment:
Changes implemented. Ready for your review.

Reply:
The domain can write like this, because the result have the same domain as the first argument.
```rust
calc_domain: Box::new(|_, args_domain| FunctionDomain::Domain(args_domain[0].clone()))
```